### ISSUE TITLE
Adding rule to ignore particular function

### ISSUE THREAD

**Issue by tprebenda** [eslint/eslint#13516]
**Please describe what the rule should do:**
The rule should allow you to turn off eslint errors that trigger on a particular function call.

For example, we are using the scriptr.io environment to perform our backend javascript functionality. Scriptr has their own unique 'require' function for importing files/functions from elsewhere in the codebase. Whenever eslint runs on a file, we get the following error:
**error  'require' is not defined       no-undef**

The only way to avoid this is to disable eslint using inline comments with every require statement. This is quite annoying, especially since we often use require statements throughout our scripts. (Refactoring all these statements to the top of the script and placing inline eslint disable comments around them is an option, though it would be very tedious and time-consuming considering the size of our code base.)

If we could just disable the eslint check of the scriptr require function in our eslint config, it would save a tremendous amount of time, and ensure that we don't accidentally disable other important eslint checks while trying to avoid errors for the require statement. 

**What new ECMAScript feature does this rule relate to?**

Unsure...

**What category of rule is this? (place an "X" next to just one item)**

[ ] Warns about a potential error (problem)
[X] Suggests an alternate way of doing something (suggestion)
[ ] Other (please specify:)

**Provide 2-3 code examples that this rule will warn about:**

```js
  let Model         = require('modules/firebase/model').FirebaseModel;
  let timeUtils     = require('utils/time');
  let modelId       = request.parameters.modelId;


  let moment 					= require("modules/moment-timezone/moment-timezone.js").moment;
  let Firebase        = require('modules/firebase/firebaseClient').Firebase;
  let Model           = require('modules/firebase/model').FirebaseModel;
  let firebaseClient  = new Firebase();
  let modelId       	= request.parameters.modelId;  //-M0UEAbLIVKt2jhJMeqP
  let targetModel     = request.parameters.targetModel; //'customer' //user'
  let model           = new Model(targetModel, modelId);
  let Campaign  			= require('models/campaign.js').Campaign;
```

**Why should this rule be included in ESLint (instead of a plugin)?**
Because this will be useful to any and all developers who are using unique environments or functions that do not fit in with the standard javascript ruleset. 

**Are you willing to submit a pull request to implement this rule?**
I would, but I don't know how.

**Comment by mdjermanovic** [comment]
Hi @tprebenda, thanks for the issue!

You can [specify globals](https://eslint.org/docs/user-guide/configuring#specifying-globals) for this environment in your eslint configuration file:

```json
{
    "globals": { 
        "require": "readonly"
    }
}
```

**Comment by tprebenda** [comment]
Hello @mdjermanovic,
Thank you so much!! I didn't even think of that. Worked like a charm!
(My question, after doing a little research, is whether or not it would be better to just set amd: true in env? I see that they also allow 'require' statements. Perhaps this is just better suited to my issue, since I'm assuming I don't need the amd env for anything else?)
Thanks again,
tprebenda

**Comment by mdjermanovic** [comment]
Glad it helped!

I'm not familiar with scriptr, so I don't know what other globals are available there. We're taking the [pre-defined environments](https://eslint.org/docs/user-guide/configuring#specifying-environments) from the [globals](https://github.com/sindresorhus/globals) package, you can see the list of variables [here](https://github.com/sindresorhus/globals/blob/master/globals.json) and it's regularly updated.

If it isn't really any of these environments, by enabling an environment you might also enable some variables that don't exist in your case, and that could cause false negatives for the `no-undef` rule , so it's probably better to define the list of globals in eslint config (`"globals"` key, as in the previous comment) or create an [environment in a plugin](https://eslint.org/docs/developer-guide/working-with-plugins#environments-in-plugins).

**Comment by tprebenda** [comment]
I see. I think I'll just add some variables under the 'globals' key in my config file for scriptr, and call it a day then!

Thank you so much!!
